test/wizard/wizard.spec.js

```diff
@@ -0,0 +1,209 @@
+
+  // load the controller's module
+  beforeEach(module(
+    'patternfly.wizard',
+    'wizard/wizard-substep.html',
+    'wizard/wizard-step.html',
+    'wizard/wizard-review-page.html',
+    'wizard/wizard.html',
+    'form/form-group/form-group.html',
+    'test/wizard/deployment.html',
+    'test/wizard/detail-page.html',
+    'test/wizard/review-second-template.html',
+    'test/wizard/review-template.html',
+    'test/wizard/summary.html',
+    'test/wizard/wizard-container.html'
+  ));
+
+  beforeEach(inject(function (_$compile_, _$rootScope_, _$httpBackend_, _$templateCache_, _$timeout_) {
+    $compile = _$compile_;
+    $scope = _$rootScope_;
+    $rootScope = _$rootScope_;
+    $httpBackend = _$httpBackend_;
+    $templateCache = _$templateCache_;
+    $timeout = _$timeout_;
+  }));
+
+  var compileHtml = function (markup, scope) {
+    var element = angular.element(markup);
+    $compile(element)(scope);
+    scope.$digest();
+    return element;
+  };
+
+  var setupWizardScope = function () {
+    var initializeWizard = function () {
+      $scope.data = {
+        name: '',
+        description: '',
+        lorem: 'default setting',
+        ipsum: ''
+      };
+      $scope.hideIndicators = false;
+
+      $scope.secondaryLoadInformation = 'ipsum dolor sit amet, porta at suspendisse ac, ut wisi vivamus, lorem sociosqu eget nunc amet.';
+      $timeout(function () {
+        $scope.deployReady = true;
+      });
+      $scope.nextButtonTitle = "Next >";
+    };
+
+    var startDeploy = function () {
+      $timeout(function() { }, 2000);
+      $scope.deployInProgress = true;
+    };
+
+    $scope.data = {};
+
+    $scope.nextCallback = function () {
+      return true;
+    };
+
+    $scope.backCallback = function () {
+      return true;
+    };
+
+    $scope.$on("wizard:stepChanged", function (e, parameters) {
+      if (parameters.step.stepId === 'review-summary') {
+        $scope.nextButtonTitle = "Deploy";
+      } else if (parameters.step.stepId === 'review-progress') {
+        $scope.nextButtonTitle = "Close";
+      } else {
+        $scope.nextButtonTitle = "Next >";
+      }
+    });
+
+    $scope.cancelDeploymentWizard = function () {
+      $rootScope.$emit('wizard.done', 'cancel');
+    };
+
+    $scope.finishedWizard = function () {
+      $rootScope.$emit('wizard.done', 'done');
+      return true;
+    };
+    initializeWizard();
+  };
+
+  beforeEach(function () {
+    setupWizardScope();
+    var modalHtml = $templateCache.get('test/wizard/wizard-container.html');
+    element = compileHtml(modalHtml, $scope);
+    $scope.$digest();
+
+    // there are two dependent timeouts in the wizard that need to be flushed
+    $timeout.flush();
+    $timeout.flush();
+  });
+
+  it('should dispatch the cancel event on the close button click', function () {
+    var closeButton = element.find('.close');
+    $rootScope.$on('wizard.done', function (event, data) {
+      expect(data).toBe('cancel');
+    });
```

Comment:
The eventFire on line 112 is what causes the wizard.done event to be emitted.  Line 109 is hit later.